src/services/expensesService.js

```diff
@@ -0,0 +1,76 @@
+'use strict';
+
+let expenses = [];
+const { deleteItemById } = require('../utils/deleteItemById');
+const { updateItem } = require('../utils/updateItem');
+const { generateId } = require('../utils/generateId');
+
+function resetExpenses() {
+  expenses = [];
+}
+
+function getFilteredExpenses({
+  userId: queryUserId,
+  categories: queryCategories = [],
+  from: queryFrom,
+  to: queryTo,
+}) {
+  let filteredExpenses = expenses;
+
+  if (queryUserId || queryCategories.length || queryFrom || queryTo) {
+    filteredExpenses = expenses.filter(({ userId, category, spentAt }) => {
+      if (queryUserId && queryUserId !== userId) {
+        return false;
+      }
+
+      if (queryCategories.length && !queryCategories.includes(category)) {
```

Comment:
if there are no `categories` in search params we will get `undefined`
if only 1 `categories` we will get just value (....?categories=1)
And if we have many categories we will get array of values (....?categories=1&categories=2&categories=3)

![image](https://github.com/mate-academy/node_accounting-app-v2/assets/86789161/ea010564-92a2-4116-9fcf-a66739d6d01f)

Your code will crash if you have only 1 category